### PR TITLE
fix(server): return errno 104 if oauth server is drunk

### LINKED ISSUE #121: GET /profile internal server error
Looks like there's an error while performing the batch request and it somehow cascades, causing a validation error to be triggered:

```
profile-server PORT 1111-8 (out): fxa-profile-server.server.ERROR: summary {"code":500,"method":"get","errno":999,"path":"/v1/profile","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.124 Safari/537.36","t":30,"stack":"Error: code is not allowed. errno is not allowed. error is not allowed. message is not allowed. info is not allowed\n    at Object.exports.create (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/node_modules/boom/lib/index.js:21:17)\n    at Object.exports.internal (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/node_modules/boom/lib/index.js:246:92)\n    at Object.exports.badImplementation (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/node_modules/boom/lib/index.js:282:23)\n    at postValidate (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/lib/validation.js:150:26)\n    at internals.Any._validateWithOptions (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/joi/lib/any.js:430:16)\n    at root.validate (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/joi/lib/index.js:102:23)\n    at exports.response (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/lib/validation.js:159:16)\n    at /Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/lib/request.js:304:13\n    at iterate (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/node_modules/items/lib/index.js:35:13)\n    at done (/Users/zcarter/moz/fxa-local-dev/fxa-profile-server/node_modules/hapi/node_modules/items/lib/index.js:27:25)"}
```

/cc @seanmonstar

### PR DESCRIPTION
**Pull request by seanmonstar**
Closes #121

@zaach interested in r?

### CODE REVIEW

**Comment by seanmonstar** [comment]
Woops, didn't update shrinkwrap.

File: lib/routes/email.js [hunk 1]
```diff
@@ -15,9 +17,13 @@ module.exports = {
     }
   },
   handler: function email(req, reply) {
-    reply({
-      email: req.auth.credentials.email
-    });
+    if (req.auth.credentials.email) {
+      reply({
+        email: req.auth.credentials.email
```
Comment (zaach) [inline lib/routes/email.js]: Nit: could store `req.auth.credentials.email` in a local variable to de-dupe

**Comment by rfk** [comment]
looks like nits addressed and ready for re-review

**Comment by zaach** [comment]
:+1:

**Comment by zaach** [comment]
Looks like it needs a rebase.

**Pull request merged**